sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeed.cs

```diff
@@ -198,6 +198,8 @@ internal DataFeedDetail GetDataFeedDetail()
                 detail.Viewers.Add(viewer);
             }
 
+            SetAuthenticationProperties(detail, DataSource);
```

Comment:
We actually have three models: a public one, an internal one used by the service, and another internal one (patch) used by the service during Update.

This method makes the transform public => service.